test/e2e/suite_test.go

```diff
@@ -202,10 +208,144 @@ func createCluster(clusterName, kubeConfigPath, controlPlane, clusterContext str
 	return nil
 }
 
+//nolint:unused
 func deleteCluster(clusterName, kubeConfigPath string) error {
 	return clusterProvider.Delete(clusterName, kubeConfigPath)
 }
 
+func createKwokCluster(clusterName string, kubeConfigPath string) (clusterContext string, err error) {
+	var nodeResource = `
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    node.alpha.kubernetes.io/ttl: "0"
+  labels:
+    beta.kubernetes.io/arch: ` + runtime.GOARCH + `
+    beta.kubernetes.io/os: linux
+    kubernetes.io/arch: ` + runtime.GOARCH + `
+    kubernetes.io/hostname: ` + clusterName + `-control-plane
+    kubernetes.io/os: linux
+    kubernetes.io/role: agent
+    node-role.kubernetes.io/agent: ""
+    type: kwok
+  name: ` + clusterName + `-control-plane
+status:
+  allocatable:
+    cpu: 32
+    memory: 256Gi
+    pods: 110
+  capacity:
+    cpu: 32
+    memory: 256Gi
+    pods: 110
+  nodeInfo:
+    architecture: ` + runtime.GOOS + `
+    bootID: ""
+    containerRuntimeVersion: ""
+    kernelVersion: ""
+    kubeProxyVersion: fake
+    kubeletVersion: fake
+    machineID: ""
+    operatingSystem: linux
+    osImage: ""
+    systemUUID: ""
+  phase: Running
+`
+
+	kwokRuntime := "docker"
+	if runtime.GOOS == "linux" {
+		kwokRuntime = "binary"
+	}
+
+	// Create a kwok cluster
+	args := []string{
+		"--name=" + clusterName,
+		"create", "cluster",
+		"--quiet-pull=true",
+		"--runtime=" + kwokRuntime,
+		"--kube-authorization=true",
+	}
+	cmd := osexec.Command("kwokctl", args...)
+	err = runCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a node for cluster
+	args = []string{
+		"--name=" + clusterName,
+		"kubectl", "apply", "-f", "-",
+	}
+	cmd = osexec.Command("kwokctl", args...)
+	cmd.Stdin = strings.NewReader(nodeResource)
+	err = runCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	// Get kubeconfig of cluster
+	buf := bytes.NewBuffer(nil)
+	args = []string{
+		"--name=" + clusterName,
+		"get", "kubeconfig",
+	}
+	cmd = osexec.Command("kwokctl", args...)
+	cmd.Stdout = buf
+	err = runCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+
+	err = os.MkdirAll(filepath.Dir(kubeConfigPath), 0755)
+	if err != nil {
+		return "", err
+	}
+
+	config := buf.Bytes()
+
+	// For Docker running on Linux used 172.17.0.1 which is the Docker-host in Docker’s default-network.
+	ip := "172.17.0.1"
+	if runtime.GOOS != "linux" {
```

Comment:
In line 258, we set the `binary` runtime for linux,  I am confused why use docker here?

Reply:
This is actually just to get the ip of the host accessing from the container, now it's really easy to misunderstand, I'll change it.